drivers/hwinfo/hwinfo_shell.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2019 Alexander Wachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <misc/printk.h>
+#include <shell/shell.h>
+#include <hwinfo.h>
+#include <zephyr/types.h>
+#include <logging/log.h>
+
+static int cmd_get_device_id(const struct shell *shell, size_t argc, char **argv)
+{
+	u8_t dev_id[16];
+	ssize_t length;
+	int i;
+
+	length = hwinfo_get_device_id(dev_id, sizeof(dev_id));
+
+	if (length == -ENOTSUP) {
+		shell_fprintf(shell, SHELL_ERROR, "Not supported by hardware\n");
+		return 0;
+	} else if (length < 0) {
+		shell_fprintf(shell, SHELL_ERROR, "Error: %d\n", length);
+		return 0;
+	}
+
+	shell_fprintf(shell, SHELL_NORMAL, "Length: %d\n", length);
```

Comment:
You can use macro: `shell_print`

`shell_print(shell, "Length: %d", length);`

Reply:
@jarz-nordic the problem with shell_print is the "\n" that is added at the end of the string